### PR TITLE
Add media service upload and download operations.

### PR DESCRIPTION
**Pull request by dewniMW**
Issue: https://github.com/wso2/product-is/issues/8499

### CODE REVIEW

File: components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/file/FileBasedStorageSystemImpl.java [hunk 1]
```diff
@@ -104,84 +123,75 @@ public void deleteFile(String id, String type, String tenantDomain) throws Stora
...
-        if (imagesPath != null) {
-            Path targetLocation = imagesPath.resolve(uuid);
-            try (FileOutputStream fileOutputStream = new FileOutputStream(targetLocation.toFile());
-                    FileChannel fileChannel = fileOutputStream.getChannel();
-                    ReadableByteChannel readableByteChannel = Channels.newChannel(fileInputStream)) {
+        if (mediaStoragePath != null) {
+            Path targetLocation = mediaStoragePath.resolve(uuid);
+            File file = targetLocation.toFile();
+            // Currently, only single file upload is allowed.
+            try (FileOutputStream fileOutputStream = new FileOutputStream(file);
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/file/FileBasedStorageSystemImpl.java:149]: shouldn't we close the streams once closed? better to use try-with resources.

**Comment by emswbandara** [comment]
shall we do a major release (2.0.0) on this repo since we are changing the package names? WDYT @dewniMW @malithie @mefarazath

File: components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/model/MediaMetadata.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        this.fileName = fileName;
+    }
+
+    public ArrayList<String> getFileIdentifiers() {
+
+        return fileIdentifiers;
+    }
+
+    public void setFileIdentifiers(ArrayList<String> fileIdentifiers) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/model/MediaMetadata.java:69]: change to List<String fileIdentifiers>

**Review by emswbandara — APPROVED** [review]

File: components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/file/FileBasedStorageSystemImpl.java [hunk 3]
```diff
@@ -0,0 +1,350 @@
+/*
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/file/FileBasedStorageSystemImpl.java:1]: Please note that due to a buggy behaviour in github, this file is shown as a completely added new file. To see only the actual changes, refer commit: https://github.com/wso2/identity-content-repository/pull/7/commits/f7766db8bd0d2da07ca5e025baf05fe4a1ce036b

File: components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/model/MediaMetadata.java (hunk 2 above)
Comment (dewniMW) [reply components/org.wso2.carbon.identity.media/src/main/java/org/wso2/carbon/identity/media/model/MediaMetadata.java]: fixed with: https://github.com/wso2/identity-content-repository/pull/7/commits/0a86f5b32f5643ca3de64b83f23a38d65e698663

**Pull request merged**